File: DatasetStats/SummaryStatDetails.vb

Namespace DatasetStats
    Public Class SummaryStatDetails
        ''' <summary>
        ''' Scan count
        ''' </summary>
        Public Property ScanCount As Integer

        ''' <summary>
        ''' Max TIC
        ''' </summary>
        Public Property TICMax As Double

        ''' <summary>
        ''' Max BPI
        ''' </summary>
        Public Property BPIMax As Double

        ''' <summary>
        ''' Median TIC
        ''' </summary>
        Public Property TICMedian As Double

        ''' <summary>
        ''' Median BPI
        ''' </summary>
        Public Property BPIMedian As Double

        ''' <summary>
        ''' Constructor
        ''' </summary>
        Public Sub New()
            Clear()
        End Sub

        Public Sub Clear()
            ScanCount = 0
            TICMax = 0
            BPIMax = 0
            TICMedian = 0
            BPIMedian = 0
        End Sub

        Public Overrides Function ToString() As String
            Return "ScanCount: " & ScanCount
        End Function
    End Class
End Namespace

File: DatasetStats/SummaryStatDetails.cs
namespace MASIC.DatasetStats
{
    /// <summary>
    /// Summary stat details
    /// </summary>
    public class SummaryStatDetails
    {
        /// <summary>
        /// Scan count
        /// </summary>
        public int ScanCount { get; set; }

        /// <summary>
        /// Max TIC
        /// </summary>
        public double TICMax { get; set; }

        /// <summary>
        /// Max BPI
        /// </summary>
        public double BPIMax { get; set; }

        /// <summary>
        /// Median TIC
        /// </summary>
        public double TICMedian { get; set; }

        /// <summary>
        /// Median BPI
        /// </summary>
        public double BPIMedian { get; set; }

        /// <summary>
        /// Constructor
        /// </summary>
        public SummaryStatDetails()
        {
            Clear();
        }

        /// <summary>
        /// Reset stored values to 0
        /// </summary>
        public void Clear()
        {
            ScanCount = 0;
            TICMax = 0;
            BPIMax = 0;
            TICMedian = 0;
            BPIMedian = 0;
        }

        /// <summary>
        /// Show the scan count
        /// </summary>
        public override string ToString()
        {
            return string.Format("ScanCount: {0}", ScanCount);
        }
    }
}
